python/ray/data/dataset.py

```diff
@@ -1561,12 +1561,25 @@ def union(self, *other: List["Dataset[T]"]) -> "Dataset[T]":
             tasks: List[ReadTask] = []
             block_partition_refs: List[ObjectRef[BlockPartition]] = []
             block_partition_meta_refs: List[ObjectRef[BlockMetadata]] = []
+
+            # Gather read task names from input blocks of unioned Datasets,
+            # and concat them before passing to resulting LazyBlockList
+            read_task_names = []
+            self_read_name = self._plan._in_blocks._read_stage_name or "read"
+            read_task_names.append(self_read_name)
+            other_read_names = [
+                o._plan._in_blocks._read_stage_name or "read" for o in other
+            ]
+            read_task_names.extend(other_read_names)
+            read_task_names.append("union")
```

Comment:
is it better if the name is `Union(ReadCSV, ReadCSV)`, instead of `ReadCSV->ReadCSV->Union`?

Reply:
Good point, let me update this naming schema.